### DISCUSSION TITLE
What would be the best way to get name, ID, and group for all students in only one section of a class?

### DISCUSSION THREAD

**Discussion by ecmagnuson** [ucfopen/canvasapi#590] [Q&A] [answered]
I am interested in organizing downloads for all students in one section based upon their group.

I have a course object

```course = canvas.get_course(123456)```

I can do something like:

```
@dataclass
class Student: 
    name: str
    ID: str
    group: str

def get_course_students(course):
    students = []
    groups = course.get_groups()
    for group in groups:
        for user in group.get_users():
            students.append(Student(user.name, user.id, group.name))
    return students
```

to get a list of all of the students (name, id, group), but this takes a while and is excessive since I only need the students from one section. 

Likewise, it doesn't actually give me any information regarding what section the students are in. 

From the API I see I can get get a `section` object:

```
section = course.get_section(12345)
```

and I can get the users in that section

```
def get_section_students(section):
    students = []
    enrollments = section.get_enrollments()
    for enrollment in enrollments:
        if enrollment.role == "StudentEnrollment":
            name = enrollment.user["name"]
            ID = enrollment.user["id"]
            group = enrollment.user["group"]  #KeyError, no key "group" 
            students.append(Student(name, ID, "no group?")
```

But this has the opposite problem. I get all students in one section, but I don't get any information on their group. 

Right now, for a quick workaround I am getting all the students and then finding the intersection between all students and only the students I want from a `section`, which works but is quite inneficient. 

Is there some better way that I am missing?

Thank you for the excellent library! It has been tremendously helpful to me.

**Comment by bennettscience** [comment, marked as answer]
You can get the group information if you use the `include` kwarg in your sections approach (which is what I would do):

```python
enrollments = section.get_enrollments(include='group_ids')
```

Any value from the Canvas REST API docs can get passed as a `**kwarg` to include more data.

**Reply by ecmagnuson** [reply]
That works so much better, thank you
